pkg/plugin/plugin.go

```diff
@@ -99,24 +105,33 @@ func (p *plugin) GenerateConfig(ctx context.Context, cs *api.OpenShiftManagedClu
 
 func (p *plugin) GenerateARM(ctx context.Context, cs *api.OpenShiftManagedCluster, isUpdate bool) ([]byte, error) {
 	log.Info("generating arm templates")
-	generator := arm.NewSimpleGenerator(p.entry)
-	return generator.Generate(ctx, cs, isUpdate)
+	return p.armGenerator.Generate(ctx, cs, isUpdate)
 }
 
 func (p *plugin) InitializeCluster(ctx context.Context, cs *api.OpenShiftManagedCluster) error {
 	log.Info("initializing cluster")
-	initializer := initialize.NewSimpleInitializer(p.entry, p.config)
-	return initializer.InitializeCluster(ctx, cs)
+	return p.clusterUpgrader.InitializeCluster(ctx, cs)
 }
 
 func (p *plugin) HealthCheck(ctx context.Context, cs *api.OpenShiftManagedCluster) error {
 	log.Info("starting health check")
-	healthChecker := healthcheck.NewSimpleHealthChecker(p.entry, p.config)
-	return healthChecker.HealthCheck(ctx, cs)
+	return p.healthChecker.HealthCheck(ctx, cs)
 }
 
-func (p *plugin) Update(ctx context.Context, cs *api.OpenShiftManagedCluster, azuredeploy []byte) error {
-	log.Info("starting update")
-	upgrader := upgrade.NewSimpleUpgrader(p.entry, p.config)
-	return upgrader.Update(ctx, cs, azuredeploy, p.config)
+func (p *plugin) CreateOrUpdate(ctx context.Context, cs *api.OpenShiftManagedCluster, azuredeploy []byte, isUpdate bool, deployFn api.DeployFn) error {
+	var err error
+	if isUpdate {
+		log.Info("starting update")
+		err = p.clusterUpgrader.Update(ctx, cs, azuredeploy, deployFn)
+		if err != nil {
+			return err
+		}
+	} else {
+		log.Info("starting deploy")
+		err := p.clusterUpgrader.Deploy(ctx, cs, azuredeploy, deployFn)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
```

Comment:
call WaitForInfraServices and console check here?

Reply:
done